# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium-shim/patches/with_update/mingw_undefine_MemoryBarrier.patch
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium-shim/patches/with_update/mingw_undefine_MemoryBarrier.patch
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/security/sandbox/chromium-shim/patches/with_update/mingw_undefine_MemoryBarrier.patch	Wed Jun 12 11:10:48 2019 +0100
@@ -0,0 +1,52 @@
+# HG changeset patch
+# User Bob Owen <bobowencode@gmail.com>
+# Date 1558340081 -3600
+#      Mon May 20 09:14:41 2019 +0100
+# Node ID ba49022852e0cd13ddb6afa501b61c1f603d60b7
+# Parent  672e222dce87dd64c3e2a1381d16aec71b21aef6
+Bug 1552160: always undefine MemoryBarrier in Windows sandbox on MinGW.
+
+diff --git a/security/sandbox/chromium/base/atomicops.h b/security/sandbox/chromium/base/atomicops.h
+--- a/security/sandbox/chromium/base/atomicops.h
++++ b/security/sandbox/chromium/base/atomicops.h
+@@ -34,17 +34,17 @@
+ // identify the STL implementation.
+ // - libc++: captures __config for _LIBCPP_VERSION
+ // - libstdc++: captures bits/c++config.h for __GLIBCXX__
+ #include <cstddef>
+ 
+ #include "base/base_export.h"
+ #include "build/build_config.h"
+ 
+-#if defined(OS_WIN) && defined(ARCH_CPU_64_BITS)
++#if defined(OS_WIN) && (defined(ARCH_CPU_64_BITS) || defined(__MINGW32__))
+ // windows.h #defines this (only on x64). This causes problems because the
+ // public API also uses MemoryBarrier at the public name for this fence. So, on
+ // X64, undef it, and call its documented
+ // (http://msdn.microsoft.com/en-us/library/windows/desktop/ms684208.aspx)
+ // implementation directly.
+ #undef MemoryBarrier
+ #endif
+ 
+diff --git a/security/sandbox/chromium/base/atomicops_internals_x86_msvc.h b/security/sandbox/chromium/base/atomicops_internals_x86_msvc.h
+--- a/security/sandbox/chromium/base/atomicops_internals_x86_msvc.h
++++ b/security/sandbox/chromium/base/atomicops_internals_x86_msvc.h
+@@ -9,17 +9,17 @@
+ 
+ #include "base/win/windows_types.h"
+ 
+ #include <intrin.h>
+ 
+ #include "base/macros.h"
+ #include "build/build_config.h"
+ 
+-#if defined(ARCH_CPU_64_BITS)
++#if defined(ARCH_CPU_64_BITS) || defined(__MINGW32__)
+ // windows.h #defines this (only on x64). This causes problems because the
+ // public API also uses MemoryBarrier at the public name for this fence. So, on
+ // X64, undef it, and call its documented
+ // (http://msdn.microsoft.com/en-us/library/windows/desktop/ms684208.aspx)
+ // implementation directly.
+ #undef MemoryBarrier
+ #endif
+ 